apps/shell/tash_command.c

```diff
@@ -564,22 +567,27 @@ int tash_execute_cmd(char **args, int argc)
 				pmdrv_fd = open("/dev/pm", O_WRONLY);
 				if (pmdrv_fd < 0) {
 					shdbg("open /dev/pm failed(%d), \n", get_errno());
+					ret = ERROR;
 				} else {
-					pm_suspended = ioctl(pmdrv_fd, PMIOC_SUSPEND, PM_NORMAL);
-					if (pm_suspended != OK) {
+					if (ioctl(pmdrv_fd, PMIOC_DOMAIN_REGISTER, &pm_domain_arg) != OK) {
+						shdbg("pm_domain_register failed(%d)\n", get_errno());
+						ret = ERROR;
+					} else if (ioctl(pmdrv_fd, PMIOC_SUSPEND, pm_domain_arg.domain_id) != OK) {
```

Comment:
we need to check pm_domain_arg.domain_id > 0

Reply:
No, if domain_id < 0, then ioctl(PMIOC_DOMAIN_REGISTER) return -1, so we don't call ioctl(PMIOC_SUSPEND) in else-if part